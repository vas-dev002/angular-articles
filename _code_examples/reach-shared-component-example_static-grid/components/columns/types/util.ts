import { Type } from "@angular/core";

import { cloneDeep } from "lodash";

import { ITypedController } from "../entity";
import { TextFactorySG } from "./text/text.factory";
import { ColumnDisplayType } from "./base-api-types";
import { IReadinessObject } from "../../../../util/entity";
import { IGroupKeyReadinessOptions } from "./readiness-edit/entity";
import { IColDefAbstractFactory } from "./base/base-abstract-factory/models";

export function mergeTypedControllersArrays(
	typedControllers1: ITypedController[],
	typedControllers2: ITypedController[]): ITypedController[] {

	const result = cloneDeep(typedControllers1);

	typedControllers2.forEach(cType => {
		const existedDef = result.find(column => column.key === cType.key);
		if (!existedDef) {
			result.push(cType);
		} else {
			existedDef.factory = cType.factory;
		}
	});

	return result;
}

export function getControllerByApiType(
	factory: ColumnDisplayType,
	typedControllers: ITypedController[]): Type<IColDefAbstractFactory> {

	const controller = typedControllers.find(column => column.key === factory);
	return (controller) ? controller.factory : TextFactorySG;
}

export function getObjGroup(readinessOptions: IReadinessObject, isMultiType: boolean): IGroupKeyReadinessOptions {
	const result: IGroupKeyReadinessOptions = {
		backColorHtml: readinessOptions.backColorHtml,
		ragStatus: getRagStatus(readinessOptions),
		isSentenceCase: readinessOptions.isSentenceCase
	};
	if (isMultiType) {
		result.stringValueMode = isMultiType,
		result.value = getRagStatus(readinessOptions);
	}
	return result;
}

export function getRagStatus(readinessOptions: IReadinessObject): string {
	let ragStatus = "";

	if (readinessOptions && readinessOptions.ragStatus) {
		ragStatus = (readinessOptions.isSentenceCase)
			? readinessOptions.ragStatus.toString()
			: readinessOptions.ragStatus.toString().toUpperCase();
	}

	return ragStatus;
}
