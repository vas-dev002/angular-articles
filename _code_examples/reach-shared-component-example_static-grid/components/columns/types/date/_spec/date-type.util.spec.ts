import { DateDasType } from "../date-type.util";

declare const expect: jest.Expect;

describe("DateDasType Utils: ", () => {

	const possibleValues = [null, undefined, "", "null", "2019-02-08T00:00:00"];

	it("[Function: getValue() should return a tooltip string]", () => {
		const result: { [key: number]: string } = {};
		possibleValues.forEach((val, index) => {
			result[index] = DateDasType.getValue(val);
		});
		expect(result).toMatchSnapshot();
	});

});
