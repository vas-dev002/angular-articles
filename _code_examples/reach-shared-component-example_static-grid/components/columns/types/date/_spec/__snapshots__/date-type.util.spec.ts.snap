// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateDasType Utils:  [Function: getValue() should return a tooltip string] 1`] = `
Object {
  "0": "",
  "1": "",
  "2": "",
  "3": "",
  "4": "Feb 8, 2019",
}
`;
